parlai/tasks/dream/agents.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import FixedDialogTeacher
+from .build import build
+import os
+import json
+
+
+def setup_data(opt, jsons_path):
+    if opt['datatype'].startswith('test'):
+        dpath = os.path.join(jsons_path, 'test.json')
+    elif opt['datatype'].startswith('valid'):
+        dpath = os.path.join(jsons_path, 'dev.json')
+    else:
```

Comment:
Would be safer to add an `elif opt['datatype'].startswith('train')` and then an `else: ValueError` here, especially if we add a `mock` fold later on